### PR TITLE
[System.Reflection.Emit] Use MonoError in static functions

### PR DESCRIPTION
**Pull request by lambdageek**

### CODE REVIEW

File: mono/metadata/reflection.c [hunk 1]
```diff
@@ -1037,9 +1056,11 @@ method_builder_encode_signature (MonoDynamicImage *assembly, ReflectionMethodBui
 			modreq = mono_array_get (mb->param_modreq, MonoArray*, i);
 		if (mb->param_modopt && (i < mono_array_length (mb->param_modopt)))
 			modopt = mono_array_get (mb->param_modopt, MonoArray*, i);
-		encode_custom_modifiers (assembly, modreq, modopt, &buf);
+		encode_custom_modifiers (assembly, modreq, modopt, &buf, error);
+		if (!is_ok (error)) goto leave;
```
Comment (ludovic-henry) [inline mono/metadata/reflection.c]: the common Mono style is to always put the body of the `if` on a new line

File: mono/metadata/reflection.c [hunk 2]
```diff
@@ -1011,7 +1028,7 @@ method_builder_encode_signature (MonoDynamicImage *assembly, ReflectionMethodBui
 	guint32 nparams =  mb->parameters ? mono_array_length (mb->parameters): 0;
 	guint32 ngparams = mb->generic_params ? mono_array_length (mb->generic_params): 0;
 	guint32 notypes = mb->opt_types ? mono_array_length (mb->opt_types): 0;
-	guint32 idx;
+	guint32 idx = 0;
```
Comment (ludovic-henry) [inline mono/metadata/reflection.c]: that's not needed as we initialise idx at line 1035.

**Comment by ludovic-henry** [comment]
LGTM

**Comment by lambdageek** [comment]
Thanks for reviewing this pile of code @ludovic-henry :+1: 

@monojenkins merge

**Pull request merged**